src/db.c

```diff
@@ -462,90 +455,46 @@ long long emptyDb(int dbnum, int flags, void(callback)(dict*)) {
     return removed;
 }
 
-/* Store a backup of the database for later use, and put an empty one
- * instead of it. */
-dbBackup *backupDb(void) {
-    dbBackup *backup = zmalloc(sizeof(dbBackup));
+/* Initialize temporary db on replica for use during diskless replication */
+tempDb *initTempDb(void) {
+    tempDb *tempDb = zmalloc(sizeof(tempDb));
 
-    /* Backup main DBs. */
-    backup->dbarray = zmalloc(sizeof(redisDb)*server.dbnum);
+    tempDb->dbarray = zmalloc(sizeof(redisDb)*server.dbnum);
     for (int i=0; i<server.dbnum; i++) {
-        backup->dbarray[i] = server.db[i];
```

Comment:
Thanks, will do

Reply:
Trying understand better (sorry the lack of C skills)
`blocking_keys, watched_keys, ready_keys` are all dictionaries the same way as dict and expires, but only dict and expires are swapped in the end.
Does it make sense to init `dict` and `expires` with NULL and then `blocking_keys, watched_keys, ready_keys` with `dictCreate`?

Reply:
you can either just replace the `tempDb->dbarray = zmalloc()` with `calloc` (zeros the memory it allocates).
or, inside your loop over `server.dbnum`, you can `memset(&tempDb->dbarray[i], 0, sizeof(redisDb));`.

in theory, we should have created an array that only holds the `dict` and `expires` members, but since we pass that `redisDb*` to functions that work with that type, we can't.
so i'm just trying to make sure we have NULL pointers there, and not random (uninitialized) ones.